frontend/javascripts/viewer/geometries/plane.ts

```diff
@@ -145,45 +153,46 @@ class Plane {
     }
     this.lastScaleFactors[0] = xFactor;
     this.lastScaleFactors[1] = yFactor;
-
-    const scaleVec = new THREE.Vector3().multiplyVectors(
-      new THREE.Vector3(xFactor, yFactor, 1),
-      this.baseScaleVector,
-    );
-    this.plane.scale.copy(scaleVec);
-    this.TDViewBorders.scale.copy(scaleVec);
-    this.crosshair[0].scale.copy(scaleVec);
-    this.crosshair[1].scale.copy(scaleVec);
+    // Account for the dataset scale to match one world space coordinate to one dataset scale unit.
+    const scaleVector: Vector3 = [
+      xFactor * this.datasetScaleFactor[0],
+      yFactor * this.datasetScaleFactor[1],
+      1 * this.datasetScaleFactor[2],
+    ];
+    this.getMeshes().map((mesh) => mesh.scale.set(...scaleVector));
   }
 
+  setBaseRotation = (rotVec: THREE.Euler): void => {
+    this.baseRotation.copy(rotVec);
+    this.baseRotationMatrix.makeRotationFromEuler(this.baseRotation);
+  };
+
   setRotation = (rotVec: THREE.Euler): void => {
-    [this.plane, this.TDViewBorders, this.crosshair[0], this.crosshair[1]].map((mesh) =>
-      mesh.setRotationFromEuler(rotVec),
-    );
+    // rotVec must be in "ZYX" order as this is how the flycam operates (see flycam_reducer setRotationReducer)
+    this.flycamRotationMatrix.makeRotationFromEuler(rotVec);
+    const combinedMatrix = this.flycamRotationMatrix.multiply(this.baseRotationMatrix);
+    this.getMeshes().map((mesh) => mesh.setRotationFromMatrix(combinedMatrix));
   };
 
   // In case the plane's position was offset to make geometries
   // on the plane visible (by moving the plane to the back), one can
-  // additionally pass the originalPosition (which is necessary for the
+  // additionally pass the offset of the position (which is necessary for the
   // shader)
-  setPosition = (pos: Vector3, originalPosition?: Vector3): void => {
-    const [x, y, z] = pos;
-    this.TDViewBorders.position.set(x, y, z);
-    this.crosshair[0].position.set(x, y, z);
-    this.crosshair[1].position.set(x, y, z);
-    this.plane.position.set(x, y, z);
-
-    if (originalPosition == null) {
-      // @ts-expect-error ts-migrate(2339) FIXME: Property 'setGlobalPosition' does not exist on typ... Remove this comment to see the full error message
-      this.plane.material.setGlobalPosition(x, y, z);
-    } else {
-      // @ts-expect-error ts-migrate(2339) FIXME: Property 'setGlobalPosition' does not exist on typ... Remove this comment to see the full error message
-      this.plane.material.setGlobalPosition(
-        originalPosition[0],
-        originalPosition[1],
-        originalPosition[2],
-      );
-    }
+  setPosition = (
+    originalPosition: Vector3,
+    positionOffset: Vector3 = DEFAULT_POSITION_OFFSET,
```

Comment:
so, 0,0,0 is always passed to the plane material factory for the 3d viewport? is that never really used?

Reply:
I had a brief look. there is no planematerialfactory for the 3d viewport. however, when the 3d viewport is rendered, setPosition is called for the xy, yz, xz viewports with an effective offset of zero, because the 3d viewport shouldn't be rendered with any offsets. so, everything makes sense and looks good :)